engine/storage_internal_stordb.go

```diff
@@ -1586,36 +1586,45 @@ func (iDB *InternalDB) GetCDRs(filter *utils.CDRsFilter, remove bool) (cdrs []*C
 			}
 		}
 
-		if filter.Paginator.Offset != nil {
-			if paginatorOffsetCounter <= *filter.Paginator.Offset {
+		if filter.OrderBy == utils.EmptyString { // if do not have to order exit early
+			if filter.Paginator.Offset != nil &&
+				paginatorOffsetCounter <= *filter.Paginator.Offset {
 				paginatorOffsetCounter++
 				continue
 			}
-		}
-		if filter.Paginator.Limit != nil {
-			if len(cdrs) >= *filter.Paginator.Limit {
+			if filter.Paginator.Limit != nil &&
+				len(cdrs) >= *filter.Paginator.Limit {
 				break
 			}
 		}
 		//pass all filters and append to slice
 		cdrs = append(cdrs, cdr)
 	}
+	if filter.OrderBy != utils.EmptyString &&
```

Comment:
I have the feeling here we could unify the code somehow not to repeat the filter part.